src/sentry/models/project.py

```diff
@@ -341,6 +342,11 @@ def transfer_to(self, team=None, organization=None):
             if not is_member:
                 rule.update(owner=None)
 
+        # null out environment_id so when the org(and its associated resources) eventually gets deleted
+        # the to_org resources aren't referencing to-be-deleted DB objects, blocking the deletion process
+        for alert_rule in list(alert_rules):
+            SnubaQuery.objects.filter(id=alert_rule.snuba_query_id).update(environment_id=None)
```

Comment:
When we look at how `Rule` objects are transferred over a few lines up it `get_or_create`s the environment in the new org and sets the environment id for the `Rule` to that environment. I think it makes sense to do the same here as well.

Reply:
Took a crack at your suggestion. Let me know what you think.